tests/test_domain_objects/cash_flow_test.py

```diff
@@ -0,0 +1,58 @@
+import sys
+from datetime import datetime, timezone
+
+sys.path.insert(0, 'tests/')
+from utils import shared_tests as shared
+from utils import helper_methods as helper
+
+
+def test_cash_flows():
+    records, domain_obj = helper.set_up_cash_flow_tests()
+    for record in records:
+        shared.attribute_quantity_valid('cash_flow', record, 7)
+        account_id_valid(record, domain_obj)
+        corporate_action_id_valid(record)
+        quantity_valid(record)
+        shared.currency_valid(record)
+        payment_status_valid(record)
+        payment_type_valid(record)
+        payment_date_valid(record)
+
+
+def account_id_valid(record, domain_obj):
+    """ account id must match to a row in the accounts database table with
+    account type of 'Client' or 'Firm'"""
+    account_id = record['account_id']
+    account_table = domain_obj.retrieve_records('accounts')
+    details_in_database = False
+    for account in account_table:
+        if account['account_id'] == account_id and \
+                account['account_type'] in ['Client', 'Firm']:
+            details_in_database = True
+            break
+    assert details_in_database
+
+
+def corporate_action_id_valid(record):
+    """ corporate_action_id must be a 10 character string """
+    account_description = record['corporate_action_id']
+    assert shared.is_length(10, account_description)
+    assert isinstance(account_description, str)
+
+
+def quantity_valid(record):
+    quantity = record['quantity']
+    assert isinstance(quantity, float)
```

Comment:
See above comment about removing this function. However - the following line is useful:

`assert isinstance(quantity, float)` 

I think this should be added to the `price_valid` method in `shared_tests.py` like so (changing `quantity` to `price`):

```
def price_valid(record, field_name='price', min=1, max=10):
    """ Random float between specified min and max values
    inclusive to 2 d.p """
    price = record[field_name]
    assert isinstance(price, float)
    string_price = str(price)
    decimal = string_price.split(".")[1]
    assert min <= price <= max and len(decimal) <= 2 
```